source/authentication/apple.txt

```diff
@@ -59,379 +59,607 @@ following:
 
-      #. On the :guilabel:`Configure Key` page, select the App ID that you
-         created as the :guilabel:`Primary App ID` and then click
-         :guilabel:`Save`.
+            .. figure:: /images/apple-team-id.png
+               :alt: An Apple team ID in the Apple Developer Portal
 
-         .. figure:: /images/apple-auth-configure-key.png
-            :alt: The Configure Key page in the Apple Developer Portal
+            Once you've confirmed that you have all the required information, you can use
+            a script to generate the JWT. You may define your own script or use the
+            script in this step.
 
-      #. Click :guilabel:`Continue` to review your key configuration. When
-         you're sure that you've configured the key correctly, click
-         :guilabel:`Register`.
+            To generate the JWT, we'll use the :rubygems:`jwt <gems/jwt>` gem. To install
+            it, run the following:
 
-      #. Copy the :guilabel:`Key ID` value somewhere that you can access it
-         later and then click :guilabel:`Download` to download the key as a
-         ``.p8`` text file. You will use these to generate the client
-         secret.
+            .. code-block:: shell
+
+               gem install jwt
 
-         .. important:: Save the Key
+            Create a new file called ``generate_client_secret.rb`` and copy the following
+            code block into the file.
 
-            You can only download the key one time. Make sure that you save
-            the key somewhere safe in case you need it again. If you lose
-            the key, you will need to generate a new one.
+            .. code-block:: ruby
 
+               # Source: https://developer.okta.com/blog/2019/06/04/what-the-heck-is-sign-in-with-apple
+               require 'jwt'
 
-   .. step:: Create the Client Secret JWT
+               # Update these values with your app's information
+               team_id = '<Apple Team ID>'
+               client_id = '<Bundle ID>'
+               key_id = '<Key ID>'
+               key_file = '<Key File Path>'
 
-      You can now create the client secret JWT for the Apple ID
-      authentication provider. Make sure that you have the following
-      information:
+               # Define the JWT's headers and claims
+               headers = {
+               # The token must be signed with your key
+               'kid' => key_id
+               }
+               claims = {
+               # The token is issued by your Apple team
+               'iss' => team_id,
+               # The token applies to Apple ID authentication
+               'aud' => 'https://appleid.apple.com',
+               # The token is scoped to your application
+               'sub' => client_id,
+               # The token is valid immediately
+               'iat' => Time.now.to_i,
+               # The token expires in 6 months (maximum allowed)
+               'exp' => Time.now.to_i + 86400*180,
+               }
```

Comment:
```suggestion
               headers = {
                 # The token must be signed with your key
                 'kid' => key_id
               }
               claims = {
                 # The token is issued by your Apple team
                 'iss' => team_id,
                 # The token applies to Apple ID authentication
                 'aud' => 'https://appleid.apple.com',
                 # The token is scoped to your application
                 'sub' => client_id,
                 # The token is valid immediately
                 'iat' => Time.now.to_i,
                 # The token expires in 6 months (maximum allowed)
                 'exp' => Time.now.to_i + 86400*180,
               }
```

Indentation is off here

Reply:
Also, can we move this to a separate `.rb` file and literalinclude that here + in the web procedure?